src/api/form/PDFForm.ts

```diff
@@ -594,11 +594,15 @@ export default class PDFForm {
         ].filter(Boolean) as PDFOperator[];
 
         page.pushOperators(...operators);
-        page.node.removeAnnot(refOrDict);
       }
 
       this.removeField(field);
     }
+
+    for (let i = 0, len = pages.length; i < len; i++) {
+      const page = pages[i];
+      page.node.clearAnnot();
```

Comment:
I'm not that familiar with the PDF specification yet, but only widget annotations should be deleted here, right?

Reply:
That's what previous code was doing but appears to not have been sufficient.

Is there a way to distinguish the type of the annotations?

Reply:
Maybe the error was caused by the rollover and down appearances that were still in the document for some widgets?